### ISSUE TITLE
Cloudinary backend

### ISSUE THREAD

**Issue by yuval-cloudinary** [rclone/rclone#7989]
#### What is your current rclone version (output from `rclone version`)?
rclone v1.67.0-DEV

#### What problem are you are trying to solve?
Supporting migrations and integrations with the popular Cloudinary service

#### How do you think rclone should be changed to solve that?
Adding a new Cloudinary backend

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by aniel300** [comment]
can chunk upload be added for this backend? 
https://cloudinary.com/documentation/upload_images#chunked_asset_upload

**Comment by yuval-cloudinary** [comment]
> can chunk upload be added for this backend? https://cloudinary.com/documentation/upload_images#chunked_asset_upload

@aniel300 That's a good call. The Cloudinary GO SDK is taking care of this, and uploads are automatically chunked for large files - https://github.com/cloudinary/cloudinary-go/blob/2e732c9f508ee4e429277125157592d9a729511e/api/uploader/upload.go#L256

**Comment by aniel300** [comment]
how come it was not doing it when i was trying this early today?

**Comment by yuval-cloudinary** [comment]
> how come it was not doing it when i was trying this early today?

@aniel300 Was the file at least 20 MB? Can you share the environment name and file name so I can triage it in the log files?
I will also test it on my end

**Comment by aniel300** [comment]
u need my cloud name?

**Comment by yuval-cloudinary** [comment]
> u need my cloud name?

yes, cloud name and the uploaded file name. A UTC time of upload would be great

**Comment by aniel300** [comment]
![image](https://github.com/user-attachments/assets/7e6277b0-0cd8-4e83-ad36-dbe8bc53322a)
and is not specific to a file but rather in general

**Comment by aniel300** [comment]
> > how come it was not doing it when i was trying this early today?
> 
> @aniel300 Was the file at least 20 MB? Can you share the environment name and file name so I can triage it in the log files? I will also test it on my end

yes the file was bigger than 20mb

**Comment by yuval-cloudinary** [comment]
@aniel300 I checked it internally, and these days have chunked uploads supported also through the 'regular' upload.
If you check https://github.com/cloudinary/cloudinary-go/blob/2e732c9f508ee4e429277125157592d9a729511e/api/uploader/upload.go#L266C15-L266C32 you'll see that we're using a single "X-Unique-Upload-Id" with multiple 20MB chunks by utilizing "Content-Range" HTTP headers to let the know where the chunk belong to

**Comment by aniel300** [comment]
so it is working or not?

**Comment by yuval-cloudinary** [comment]
> so it is working or not?

Yes, it works. Whenever uploading a file larger than 20MB, the GO SDK splits it into 20MB chunks and uploads them through the upload (not upload_chunked) endpoint

**Comment by aniel300** [comment]
ok i will try again and report back, thank you for ur patience.
